File: 448-find-all-numbers-disappeared-in-an-array/find-all-numbers-disappeared-in-an-array.cpp
class Solution {
public:
    vector<int> findDisappearedNumbers(vector<int>& nums) {
        int n = nums.size();
        int i = 0;
        while(i < n) {
            int correctIndex = nums[i] - 1;
            if(nums[correctIndex] != nums[i]) {
                swap(nums[i], nums[correctIndex]);
            } else {
                i++;
            }
        }
        
        vector<int> ans;
        for(int j=0; j<n; j++) {
            if(nums[j] != j+1) {
                ans.push_back(j+1);
            }
        }

        return ans;
    }
};

File: 448-find-all-numbers-disappeared-in-an-array/find-all-numbers-disappeared-in-an-array.py
class Solution:
    def findDisappearedNumbers(self, nums):
        n = len(nums)
        temp = [0] * (n + 1)

        for num in nums:
            temp[num] = 1

        ans = []
        for i in range(1, n + 1):
            if temp[i] == 0:
                ans.append(i)

        return ans
